### PR TITLE
[load_gen] improve details on failures to show original request and response code and body

### PR DESCRIPTION
**Pull request by jrgm**
When running load_gen, there are occasionally errors reported, which I'm pretty sure are database replication races. I don't believe that these actually happen in production (for the most part). The load_gen tool inherently does sequences of wsapi calls that a real user would never perform at the same speed. 

However, the errors are reported with so little detail that it's not possible to trace them. So this PR increases the errors to always show the response code and body, and the request method, host and URI.

Before:

```
error: (signup) responseNotOk: non 200 response code: 400
error: (signup) unknown: null response body
```

After:

```
error: (signup) responseNotOk: non 200 response code >>> {"code":400,"body":"Bad Request: that email does not belong to you","req":{"method":"POST","host":"login.anosrep.org","path":"/wsapi/cert_key"}}
error: (signup) responseNotOk: non 200 response code >>> {"code":400,"body":"","req":{"method":"GET","host":"login.anosrep.org","path":"/wsapi/fake_verification?email=foo%40loadtest.domain"}}
```

### CODE REVIEW

File: lib/wsapi_client.js [hunk 1]
```diff
@@ -27,6 +27,23 @@ function startRequest() {
...
+    if (message._headers) {
+      req.host = message._headers.host;
+    }
+    req.path = message.path;
+
+    return JSON.parse(JSON.stringify(req));
+  }
+
+  return req;
+}
```
Comment (jrgm) [inline lib/wsapi_client.js:45]: And yeah, I am touching "private" API elements. I understand that this, theoretically, could change in the future.

File: lib/load_gen/activities/interaction_data.js [hunk 2]
```diff
@@ -43,17 +45,12 @@ var blob = { data:
...
-    } else if (!r || r.code !== 200) {
-      cb("for interaction_data post response code is not 200: " + (r ? r.code : "no response"));
-    } else if (r.body !== '{"success":true}' ){
-      cb("interaction_data post not successful");
-    } else {
-      cb();
+    err = common.checkResponse(err, r);
+    if (err) return cb(err);
+    if (r.body.success !== true) {
+      return cb(common.error("interaction_data post not successful", null, r));
```
Comment (jrgm) [inline lib/load_gen/activities/interaction_data.js:52]: common.checkResponse handles all the null response, non-200 and JSON.parse checking that I replace here.

File: bin/load_gen [hunk 3]
```diff
@@ -218,8 +218,8 @@ function poll() {
             completed[act][2]++;
           } else {
             completed[act][1]++;
-            winston.error('('+act+') ' + err.toString());
           }
+          winston.error('('+act+') ' + err.toString());
         } else {
```
Comment (jrgm) [inline bin/load_gen:223]: This change means that 503's are always printed. (I think some might be misclassified as being errors from the toobusy module, so I want to log these for a while).

**Comment by jaredhirsch** [comment]
:beers:

**Pull request merged**